module1/lessons/intro_to_how_the_web_works.md

```diff
@@ -0,0 +1,151 @@
+  - Domain - location of the resource
+3. Once inside, you follow Kat's instructions for getting from the front door, to the master bedroom, to the cat tower in the corner to find Autumn.
+  - Path - the specific location of our resource at this server
+4. Now you give Autumn her treat and a little pat on the head.
+  - Query String - additional information about the information we want or the action we want to perform.
+
+
+
+The "Domain", "Path", and "Query String" combined indicate a unique "identifier" for a resource, and all three of these pieces are a **URI**.
+
+You will be covering more specifics around these parts of the URL later in Mod 2 - for now, it is important that you know that these parts exist.
+
+## The Request and Response Cycle
+
+As you start to build out web applications in Mod 2, it is important to be able to visualize the way information flows through the system; typically called the Request/Response Cycle.
+
+First a user gives a client a URL, the client builds a **request** for information (or resources) to be generated by a server.  When the server receives that request, it uses the information included in the request to build a **response** that contains the requested information. Once built, that response is sent back to the client in the requested format, to be rendered to the user.
+
+It is our job as web developers to build out and maintain servers that can successfully build responses based on standardized requests that will be received.  But, what does a standard request look like?  We need to know that before we can start building servers that will respond successfully.
+
+The standard, or protocol we use is **HTTP**.
+
+## HTTP Requests and Responses
+
+The HyperText Transfer Protocol gives us rules about how messages should be sent around the Internet. The system that initiates a connection sends a "request", and the system the answers sends a "response".
+
+### HTTP Request
+
+When a "client" (like a web browser) retrieves information, it sends a payload of data to a server as a "request". This request has many parts, including a protocol, a path (URI), and any additional information that might be needed to perform the request. For example, if we were trying to add a new cat to our cat database, we'd need to send along the attributes of the specific cat we were creating.
+
+### HTTP Response
+
+Similar to a Request, the Response also has many parts, including information on if the request was successfully completed or not. If the resource was located and can be sent back, it will be sent in the response payload.
+
+
+## Your Turn
+
+Take turns with your group answering the questions below. Feel free to use/draw a diagram to assist you in your answer.
+
+* Describe the HTTP request/response cycle. Use each of the following terms:
+  * User
+  * URL
+  * Client
+  * Server
+  * Request
+  * Response
+  * HTTP
+* What are the parts of a URL?
```

Comment:
I love it! The only suggestion I have is to add a few links to helpful request/response resources. But, you can take that suggestion or leave it! Either way the lesson looks great.

Reply:
I was thinking about doing that, but was having them research for their warmup. I think it would be good to put some helpful resources though, good idea!